docs/DEVGUIDE.md

```diff
@@ -1,5 +1,24 @@
 # Developer's Guide to Service-Catalog
 
+## Overview
+
+This repository is organized as similarly to Kubernetes itself as the developers
+have found possible (or practical). Below is a summary of the repository's
+layout:
+
+    .
+    ├── .glide         # Glide cache (untracked)
+    ├── bin            # Destination for binaries compiled for linux/amd64 (untracked)
+    ├── build          # Contains build-related scripts and subdirectories containing Dockerfiles
+    ├── cmd            # Contains "main" Go packages for each service catalog component binary
+    ├── contrib        # Contains all non-essential source
+    ├── deploy
+    │   └── catalog    # Helm chart for deploying service catalog
+    ├── docs           # Documentation
+    ├── hack           # Contains scripts useful for development
```

Comment:
right to me I'd prefer to merge build and hack since the distinction is very thin to me - and then with a "jenkins" sub-dir for all jenkins related/specific things.

And I would prefer to move top level scripts (like run.sh) into one of those dirs.